app/features/syllabus_generator/tools.py

```diff
@@ -0,0 +1,331 @@
+USABLE_OPTIONS.pop("all", None)
+
+
+class SyllabusBuilder:
+    def __init__(
+        self,
+        subject: str,
+        grade_level: str,
+        course_overview: str = "",
+        customisation: str = "",
+        options: List[str] = ["all"],
+        prompt: str = "",
+        model=None,
+        parser=None,
+        verbose: bool = False,
+    ):
+        """Initialize SyllabusBuilder with default configurations."""
+        default_config = {
+            "model": GoogleGenerativeAI(model="gemini-1.0-pro"),
+            "parser": JsonOutputParser(pydantic_object=SyllabusModel),
+            "prompt": read_text_file("prompt/syllabus_prompt.txt"),
+        }
+
+        self._validate_inputs(subject, grade_level, options, customisation)
+
+        self.prompt = prompt or default_config["prompt"]
+        self.model = model or default_config["model"]
+        self.parser = parser or default_config["parser"]
+        self.grade_level_assessments = ""
+        self.customisation = customisation
+        self.options = list(map(lambda x: x.strip().lower(), options))
+
+        self.subject = subject
+        self.grade_level = grade_level.strip().lower()
+        self.course_overview = course_overview
+        self.verbose = verbose
+
+        self._post_validate_inputs()
+
+    def _validate_inputs(self, subject, grade_level, options, customisation):
+        """Validate inputs to ensure all required fields are provided."""
+        if subject is None or len(subject) <= 2 or not isinstance(subject, str):
+            raise ValueError("Subject must be provided as a str")
+        if (
+            grade_level is None
+            or len(grade_level) == 0
+            or not isinstance(grade_level, str)
+        ):
+            raise ValueError("Grade level must be provided as a str")
+        if options is None or len(options) == 0 or not isinstance(options, list):
+            raise ValueError("Options must be provided as a list")
+        if customisation is not None and not isinstance(customisation, str):
+            raise ValueError("customisation should be str or not specified at all")
+
+    def _post_validate_inputs(self):
+        # Guarantee options only contains "all" or doesn't contain it at all
+        seen_all = False
+        for item in self.options:
+            if seen_all:
+                raise ValueError(
+                    "Invalid options provided: 'all' option already specified"
+                )
+            if item not in ALLOWED_OPTIONS:
+                raise ValueError("Invalid options provided")
+            if item == "all":
+                seen_all = True
+
+    # Probably a better way to do this
+    # TODO: Make this return ENUM so we can attach which part is broken
+    def _validate_response(self, response: Dict) -> bool:
+        """
+        Validates response from LLM
+        """
+        try:
+            # Assuming reponse is already a dict
+            if not isinstance(response, dict):
```

Comment:
```
from pydantic import BaseModel, Field, model_validator
from typing import Optional
from typing_extensions import Annotation

class LLMOutput(BaseModel):
  title: str
  overview: str
  description: Annotation[str, Field(default="This is a default desc.")]
  assignments: Annotation[List[Assignment], Field(default_factory=list)] 
  ....
```

Pydantic for validation can be useful here for validating the output

Reply:
I couldn't find any documentation on `Annotation` but I found `Annotated`, I assume this is what you meant.
Should I use this over just `Field`  in `model.py`.